packages/client/hmi-client/src/components/dataset/tera-save-dataset-from-simulation.vue

```diff
@@ -0,0 +1,57 @@
+<template>
+	<Button
+		:disabled="isSaveDisabled"
+		outlined
+		title="Saves the current version of the model as a new Terarium asset"
+		@click="showSaveInput = !showSaveInput"
+	>
+		<span class="pi pi-save p-button-icon p-button-icon-left"></span>
+		<span class="p-button-text">Save as new dataset</span>
+	</Button>
+	<span v-if="showSaveInput" style="padding-left: 1em; padding-right: 2em">
+		<InputText v-model="saveAsName" class="post-fix" placeholder="New dataset name" />
+		<i
+			class="pi pi-times i"
+			:class="{ clear: hasValidDatasetName }"
+			@click="(saveAsName = ''), (showSaveInput = false)"
+		></i>
+		<i
+			v-if="useProjects().activeProject.value?.id"
+			class="pi pi-check i"
+			:class="{ save: hasValidDatasetName }"
+			@click="saveDatasetToProject"
+		></i>
+	</span>
+</template>
+
+<script setup lang="ts">
+import { computed, ref } from 'vue';
+import { saveDataset } from '@/services/dataset';
+import { useProjects } from '@/composables/project';
+import InputText from 'primevue/inputtext';
+import Button from 'primevue/button';
+
+const props = defineProps<{
+	simulationRunId: string;
+}>();
+
+const saveAsName = ref('');
+const showSaveInput = ref(<boolean>false);
```

Comment:
should be written as ref<boolean>(false) but just do ref(false) it already assumes its boolean

Reply:
Dont know why it was like that, good shout